client/src/components/recommended-action-button.tsx

```diff
@@ -62,102 +67,204 @@ export default function RecommendedActionButton(props: {
       });
     });
   });
+  const incompleteRequirement = categories
+    .filter((c) => c.isRequired)
+    .find((c) => c.answers.find((a) => a.status === Status.INCOMPLETE));
+  console.log;
   const firstIncomplete = categories.find((c) =>
     c.answers.find((a) => a.status === Status.INCOMPLETE)
   );
 
-  function recommend(): {
+  const completedAnswers =
+    props.mentor?.answers.filter((a) => a.status === Status.COMPLETE).length ||
+    0;
+  const totalAnswers = props.mentor?.answers.length || 0;
+
+  interface Conditions {
+    idle: Answer | undefined;
+    idleIncomplete: boolean;
+    isVideo: boolean;
+    thumbnail: string;
+    categories: category[];
+    incompleteRequirement: category | undefined;
+    firstIncomplete: category | undefined;
+    completedAnswers: number;
+    totalAnswers: number;
+  }
+  interface Recommendation {
     text: string;
     reason: string;
     action: () => void;
-  } {
-    if (mentor?.thumbnail == "")
+    skippable: boolean;
+    skip: Conditions;
+  }
+
+  function recommend(conditions: Conditions): Recommendation {
+    if (conditions.thumbnail == "")
       return {
         text: "Add a Thumbnail",
         reason: "A thumbnail helps a user identify your mentor",
         action: () => undefined,
+        skippable: true,
```

Comment:
not new, but the return type of `action` above should be `void`

Reply:
Last push addresses most changes except this one. What's the proper way to do this?

Reply:
I think can just be `action: () => void` does that not work?

Reply:
nope, doesn't seem to like that

Reply:
oh sorry that's the actual function; i thought it was a type spec. I think this is more standard:  `action: () => {}`